145 test screen

Jake-Duffy855 opened the pull request:
closes #145

shashjar committed:
Added test welcome screen, test screen, and test results screen

shashjar committed:
Test screen

Jake-Duffy855 committed:
made test results

Jake-Duffy855 committed:
Merge branch 'dev' into 145-test-screen

Jake-Duffy855 committed:
progress bar

shashjar committed:
Added words for participant test screen

shashjar committed:
In process of fixing yes/no off by one bug

Jake-Duffy855 committed:
fixed scoring

shashjar committed:
Started adding firebase type and updating logic for test answers for
each participant - still have to finish writing logic for when a test
session is ended and the firebase data needs to be updated

Jake-Duffy855 committed:
finished test metrics

Jake-Duffy855 committed:
Merge branch 'dev' into 145-test-screen

Jake-Duffy855 committed:
commented code gone

shashjar committed:
Formatting changes for PR

shashjar committed:
Merge branch '145-test-screen' of https://github.com/sandboxnu/flow-in-the-field into 145-test-screen

shashjar committed:
Disables start test button before test date has arrived

shashjar committed:
Tried to make test welcome screen responsive for small screens

emaela commented on frontend/firebase.json:
```diff
@@ -5,11 +5,11 @@
     },
     "firestore": {
-      "port": 8080,
+      "port": 8085,
```
I'm curious, why'd you guys change this?

emaela commented on frontend/src/screens/TestScreen/TestScreen.tsx:
```diff
@@ -0,0 +1,199 @@
+    const TESTWORDS = require('./testWords.json');
+    const [answer, setAnswer] = useState<string>("");
+    const [answers, setAnswers] = useState<string[]>([]);
+    const answerKey = ['no', 'yes', 'no', 'no', 'yes', 'yes', 'no', 'yes', 'yes', 'no', 'yes', 'no', 'yes', 'no', 'yes', 'yes', 'yes', 'no', 'yes', 'yes', 'yes', 'no', 'no', 'yes', 'yes', 'yes', 'no', 'yes', 'no', 'yes', 'no', 'no', 'yes', 'no', 'yes', 'yes', 'no', 'yes', 'no', 'yes', 'no', 'yes', 'no', 'yes', 'no', 'yes', 'yes', 'yes', 'yes', 'yes'];
```
This is a little nit-picky but could we split long lines like these into multiple lines for readability? I feel like this could apply to the `TouchableOpacity`s in lines 110-117, and it might be worth double checking other files as well.

emaela replied:
I'd also be interested in hearing your thoughts on possibly moving this answer key out of the code - maybe it would be appropriate to have it in our representation of the test word pairs in Firestore?

emaela reviewed (changes requested):
I changed `TestWelcomeScreen` locally and the vocabulary test looked good to me! 

The updated flex attributes on `TestWelcomeScreen` will have to be committed to this branch before it can be merged. I also added a small comment about breaking up long lines in the code.

I think it could be better to have the testWords in Firestore and possibly the answer key as well (maybe something like `{turkish, english, answer}`?) - I'm interested in hearing your guys' thoughts on this. 
For uploading these words, you guys could use a script similar to the one in `upload-script`.

shashjar committed:
Addition of test script for putting test words on firestore#

shashjar committed:
Added test words to Firestore. Test is fully functional (accessing test
words, dispalaying test words, grading participant answers, updating
data on Firestore). Test availability logic is functional (test option
on home screen will be valid only if it has been 1 week since the
participant first registered and the participant hasn't already taken
the test).

emaela commented on frontend/src/screens/TestScreen/TestScreen.tsx:
```diff
@@ -0,0 +1,210 @@
+        } else if (answers.length == numQuestions) {
+            let testGradingInfo = gradeAnswers();
+            fi.endTestRound(currentTestRoundId, determineIfRoundAnswerCorrect());
+            fi.endTestSession(testSessionId, testGradingInfo.score, testGradingInfo.correctAnswers).then(() => navigation.navigate("TestResultsScreen", { correct: testGradingInfo.score, total: numQuestions }));
```
Can we split these long lines into multiple lines?

emaela reviewed (commented):
LGTM, except when I try `yarn emulate` it fails - could someone else pull this branch and try it to see whether it's just me? @Jake-Duffy855 @Octillerysnacker @nicolesorial

emaela commented:
Also, could you guys resolve the merge conflicts so that we can test the branch with them before merging into dev?

shashjar committed:
Merge branch 'dev' into 145-test-screen

shashjar committed:
dependencies???

shashjar committed:
Broke up longer lines to make more readable

shashjar committed:
Broke up longer lines

Jake-Duffy855 committed:
changed db file to MacOS

Jake-Duffy855 committed:
Merge branch 'dev' into 145-test-screen

emaela reviewed (approved):
LGTM
